shopping_list_frontend/lib/screens/login.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_form_builder/flutter_form_builder.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:shopping_list_frontend/cubits/general/route_cubit.dart';
+import 'package:shopping_list_frontend/cubits/user/login_cubit.dart';
+import 'package:shopping_list_frontend/cubits/user/user_cubit.dart';
+
+import 'package:shopping_list_frontend/exceptions/users.dart';
+import 'package:shopping_list_frontend/services/notifiers/popups.dart';
+import 'package:shopping_list_frontend/validators/user/user_validators.dart';
+
+import 'package:shopping_list_frontend/services/lifecycle/navigator.dart';
+
+class LoginScreen extends StatelessWidget {
+  LoginScreen({super.key});
+
+  final _formKey = GlobalKey<FormBuilderState>();
+  final userCubit = UserCubit();
+
+  Future<void> _submitForm(BuildContext context) async {
+    var currentState = _formKey.currentState!;
+    if (!currentState.validate()) {
+      return;
+    }
+    Map fields = currentState.fields;
+    final response = await context.read<LoginCubit>().login(
+          username: fields["username"].value,
+          password: fields["password"].value,
+        );
+    if (response != null) {
```

Comment:
I preferred operate with SP inside some repository or cubit not inside widget, pls fix it